monix-tail/shared/src/main/scala/monix/tail/internal/IterantOnError.scala

```diff
@@ -48,18 +48,23 @@ private[tail] object IterantOnError {
       Suspend[F, A](next, stop)
     }
 
+    def tailGuard(ffa: F[Iterant[F, A]]): F[Iterant[F, A]] =
+      ffa.handleError(f)
```

Comment:
PS: gone ahead and opened a ticket for a `transformWith` in `MonadError`, fingers crossed — https://github.com/typelevel/cats/issues/2161

We can't wait for it though. It will be ages before that happens, if at all 😀

Reply:
That won't hurt. In the meantime, I think, users will appreciate correctness anyway 😄